lib/_http_server.js

```diff
@@ -262,9 +262,8 @@ function writeHead(statusCode, reason, obj) {
     // Slow-case: when progressive API and header fields are passed.
     let k;
     if (obj) {
-      const keys = ObjectKeys(obj);
-      for (let i = 0; i < keys.length; i++) {
-        k = keys[i];
+      for (const key of ObjectKeys(obj)) {
```

Comment:
We should not make this change for performance reasons.

Reply:
Resolved in https://github.com/nodejs/node/pull/30958/commits/c2f4695d1f19b5841a162c92e4e041545a782186